lib/prometheus/client/push.rb

```diff
@@ -23,6 +23,7 @@ class Push
 
       def initialize(job:, instance: nil, gateway: DEFAULT_GATEWAY, **kwargs)
         raise ArgumentError, "job cannot be nil" if job.nil?
+        raise ArgumentError, "job cannot be empty" if job.empty?
```

Comment:
Shall we do `!job || job.empty?` (you're effectively reimplementing `.blank?` here, i'd do literally the same thing, and in just one statement).
Opinion, ignore me if you disagree

Reply:
Not mega fussed either way. For the sake of the extra line I'm gonna keep the more specific error messages.